server/src/test/java/io/crate/integrationtests/InsertIntoIntegrationTest.java

```diff
@@ -2065,6 +2043,34 @@ ON CONFLICT (id) DO UPDATE SET value = excluded.value
         assertThat(response)
             .as("No underreplicated/failed shards due to streaming errors")
             .hasRows("0");
+
+        execute("refresh table tbl");
+        execute("select y, write_ts, z, x, x2 from tbl where id = 1");
+        Object[] updatedRow = response.rows()[0];
+        Integer updatedY = (Integer) updatedRow[0];
+        Long updatedWriteTs = (Long) updatedRow[1];
+        Double updatedZ = (Double) updatedRow[2];
+        Integer updatedX = (Integer) updatedRow[3];
+        Integer updatedX2 = (Integer) updatedRow[4];
+        assertThat(updatedY)
+            .as("Default value does not change")
+            .isEqualTo(0);
+
+        assertThat(updatedWriteTs)
+            .as("Update must have re-evaluated the generated expression")
+            .isGreaterThan(writeTs);
+
+        assertThat(updatedZ)
+            .as("Undeterministic default value is not re-evaluated")
```

Comment:
I'm confused with this, the updatedZ shouldn't be re-evaluated, right?
(that's why you assert it against previous value `z`)